File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(number):
    return sum((number for number in range(1, number + 1))) ** 2


def sum_of_squares(number):
    return sum(number**2 for number in range(1, number + 1))


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)

File: julia/difference-of-squares/difference-of-squares.jl
"Square the sum of the first `n` positive integers"
function square_of_sum(n)
    sum(1:n)^2
end

"Sum the squares of the first `n` positive integers"
function sum_of_squares(n)
    sum([i^2 for i in 1:n])
end

"Subtract the sum of squares from square of the sum of the first `n` positive ints"
function difference(n)
    square_of_sum(n) - sum_of_squares(n)
end
